pkg/client/results/junit.go

```diff
@@ -52,3 +58,51 @@ func Passed(testCase reporters.JUnitTestCase) bool {
 func Failed(testCase reporters.JUnitTestCase) bool {
 	return testCase.Skipped == nil && testCase.FailureMessage != nil
 }
+
+func processJunitFile(pluginDir, currentFile string) (Item, error) {
+	junitResults := reporters.JUnitTestSuite{}
+	relPath, err := filepath.Rel(pluginDir, currentFile)
+	if err != nil {
+		logrus.Errorf("Error making path %q relative to %q: %v", pluginDir, currentFile, err)
+		relPath = currentFile
+	}
+
+	// Passed unless a failure is encountered for aggregate status.
+	resultObj := Item{
+		Name:     filepath.Base(currentFile),
+		Status:   StatusPassed,
+		Metadata: map[string]string{"file": relPath},
+	}
+
+	infile, err := os.Open(currentFile)
+	if err != nil {
+		resultObj.Status = StatusFailed
```

Comment:
I'm not sure if the status should be `Failed` if we fail to open the file or decode it as there could be non-test related reasons e.g. incorrect file permissions or something. Perhaps `StatusUnknown` would work for these cases? I don't feel very strongly about it though, so if you had particular reasons for using that status that's fine.